ConnectorArchicad/ConnectorArchicad/Converters/Converters/BeamConverter.cs

```diff
@@ -42,17 +42,19 @@ CancellationToken token
 
             // upgrade (if not Archicad beam): Objects.BuiltElements.Beam --> Objects.BuiltElements.Archicad.ArchicadBeam
             {
-              var baseLine = (Line)beam.baseLine;
-              var newBeam = new Objects.BuiltElements.Archicad.ArchicadBeam
+              if (beam.baseLine is Line baseLine)
               {
-                id = beam.id,
-                applicationId = beam.applicationId,
-                archicadLevel = Archicad.Converters.Utils.ConvertLevel(beam.level),
-                begC = Utils.ScaleToNative(baseLine.start),
-                endC = Utils.ScaleToNative(baseLine.end)
-              };
-
-              beams.Add(newBeam);
+                beams.Add(
+                  new Objects.BuiltElements.Archicad.ArchicadBeam
+                  {
+                    id = beam.id,
+                    applicationId = beam.applicationId,
+                    archicadLevel = Archicad.Converters.Utils.ConvertLevel(beam.level),
+                    begC = Utils.ScaleToNative(baseLine.start),
+                    endC = Utils.ScaleToNative(baseLine.end)
+                  }
+                );
+              }
```

Comment:
I'm assuming from this that if a `Beam` doesn't have a line (has something else) we will silently fail?
At least something should be added to the report in DUI I guess?

Reply:
@AlanRynne You are right about reporting. In these cases we silently fail and the report will be "Unknown". However currently the reporting is implemented on C++ side, so I think this should be another ticket (it is not a regression).

Reply:
Ticket added: https://spockle.atlassian.net/browse/CNX-9060